politeiad/cmd/legacypoliteia/convert.go

```diff
@@ -0,0 +1,1064 @@
+		if limit == 0 {
+			cvd.Address = addrs[cvj.CastVote.Ticket]
+			cvd.Timestamp = ts[cvj.CastVote.Token][cvj.CastVote.Ticket]
+
+			// Verify cast vote details signature
+			err = client.CastVoteDetailsVerify(convertCastVoteDetailsToV1(cvd),
+				serverPubkey)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		// Save the cast vote
+		votes[cvd.Ticket] = cvd
+
+		printInPlace(fmt.Sprintf("    Vote %v", count))
+
+		// If the number of converted cast votes hits the limit, we are done
+		if len(votes) == limit {
+			break
+		}
+	}
+	err = scanner.Err()
+	if err != nil {
+		return nil, err
+	}
+	fmt.Printf("\n")
+
+	// Put the votes into a slice and sort them by timestamp,
+	// from oldest to newest.
+	castVotes := make([]ticketvote.CastVoteDetails, 0, len(votes))
+	for _, vote := range votes {
+		castVotes = append(castVotes, vote)
+	}
+	sort.SliceStable(castVotes, func(i, j int) bool {
+		return castVotes[i].Timestamp < castVotes[j].Timestamp
+	})
+
+	return castVotes, nil
+}
+
+// convertCastVoteDetailsToV1 is used to verify the cast vote details signature
+// through the www client.
+func convertCastVoteDetailsToV1(vote ticketvote.CastVoteDetails) v1.CastVoteDetails {
+	return v1.CastVoteDetails{
+		Token:     vote.Token,
+		Ticket:    vote.Ticket,
+		VoteBit:   vote.VoteBit,
+		Address:   vote.Address,
+		Signature: vote.Signature,
+		Receipt:   vote.Receipt,
+		Timestamp: vote.Timestamp,
+	}
+}
+
+// decodeBallotJournalLine decodes a line from a proposal's ballot journal. The
+// line will contain two JSON structures. The JournalAction structure and the
+// CastVoteJournal structure.
+func decodeBallotJournalLine(line []byte) (*gitbe.CastVoteJournal, error) {
+	r := bytes.NewReader(line)
+	d := json.NewDecoder(r)
+
+	// Decode the action
+	var j gitbe.JournalAction
+	err := d.Decode(&j)
+	if err != nil {
+		return nil, err
+	}
+	if j.Action != gitbe.JournalActionAdd {
+		return nil, fmt.Errorf("invalid action '%v'", j.Action)
+	}
+
+	// Decode the cast vote
+	var cvj gitbe.CastVoteJournal
+	err = d.Decode(&cvj)
+	if err != nil {
+		return nil, err
+	}
+
+	return &cvj, nil
+}
+
+type commentTypes struct {
+	Adds  []comments.CommentAdd
+	Dels  []comments.CommentDel
+	Votes []comments.CommentVote
+}
+
+// likeCommentV1 unmarshals the like action data from the gitbe's comments
+// journal.
+type likeCommentV1 struct {
+	Token     string `json:"token"`               // Censorship token
+	CommentID string `json:"commentid"`           // Comment ID
+	Action    string `json:"action"`              // Up or downvote (1, -1)
+	Signature string `json:"signature"`           // Client Signature of Token+CommentID+Action
+	PublicKey string `json:"publickey"`           // Pubkey used for Signature
+	Receipt   string `json:"receipt,omitempty"`   // Signature of Signature
+	Timestamp int64  `json:"timestamp,omitempty"` // Received UNIX timestamp
+}
+
+func (c *convertCmd) convertComments(proposalDir, userID string) (*commentTypes, error) {
```

Comment:
Break the conversion of each comment type into it's own function so it can be read and audited easier.